lib/server-builder.js

```diff
@@ -44,7 +54,7 @@ class ServerBuilder {
     this.additionalAppDependencies = buildConfiguration.additionalAppDependencies || [];
   }
 
-  async build () {
+  async build (testAppPackage, signingConfig) {
```

Comment:
I'd rather put arguments into an object - it might be necessary to add more later

Reply:
perhaps, it would even make sense to pass them into the instance constructor, since they are used by multiple methods

Reply:
At first I made it in constructor. Then I saw that in constructor we also have configuration json, which defines tool versions. These version are the actual part of configuring `ServerBuilder` object. Modifying `espresso-server`'s `build.gradle` I figured out that to test an application with espresso on Android we need 1) to specify the application package which will be under test (because without it android wouldn't know what to test) and 2) to sign `espresso-server` apk with the same key as application under test (otherwise, SecurityException will occur at runtime). So, it makes specifying package the mandatory step in order to make espresso driver working. This pushed me to think that `ServerBuilder.build()` could actually be called several times with different packages and signingConfigs, but with the same tools configuration as them are defined in Android SDK and barely could change as often as package and signingConfig. So, I made it that way. What are your thoughts? Should I make it in constructor anyway?

Reply:
I would say it makes sense to put the stuff into the constructor. We can always create a new instance when necessary.